dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1021,6 +1029,102 @@ def _get_percentile(self, percentiles):
             quantiles[percentiles == 50] = median_value
         return quantiles.tolist()
 
+    @staticmethod
+    def _fold_histogram(bin_counts, bin_edges, value):
+        """
+        Offset the histogram by the given value,
+        then fold the histogram at the break point.
+
+        :param bin_counts: bin counts of the histogram
+        :type bin_counts: np.array
+        :param bin_edges: bin edges of the histogram
+        :type bin_edges: np.array
+        :param value: offset value
+        :type value: float
+        :return: two histograms represented by bin counts and bin edges
+        """
+        # normalize bin counts
+        bin_counts = bin_counts.astype(float)
+        normalized_bin_counts = bin_counts / np.sum(bin_counts)
+        bin_edges = bin_edges - value
+
+        # find the break point to fold the deviation list
+        id_zero = None
+        for i in range(len(bin_edges)):
+            if bin_edges[i] > 0:
+                id_zero = i
+                break
+
+        # if all bin edges are positive or negative (no break point)
+        if id_zero is None:
+            return [[], []], [normalized_bin_counts, bin_edges]
+        if id_zero == 1:
+            return [normalized_bin_counts, bin_edges], [[], []]
+
+        # otherwise, generate two folds of deviation
+        bin_counts_pos = np.append(
+            [normalized_bin_counts[id_zero - 1] * (bin_edges[id_zero] /
+            (bin_edges[id_zero] - bin_edges[id_zero - 1]))],
+            normalized_bin_counts[id_zero:])
+        bin_edges_pos = np.append([0], bin_edges[id_zero:])
+
+        bin_counts_neg = np.append(
+            [normalized_bin_counts[id_zero - 1] -
+            normalized_bin_counts[id_zero - 1] * (bin_edges[id_zero] /
+            (bin_edges[id_zero] - bin_edges[id_zero - 1]))],
+            normalized_bin_counts[:id_zero - 1][::-1])
+        bin_edges_neg = np.append([0], -bin_edges[:id_zero][::-1])
+
+        if len(bin_edges_neg) > 1 and bin_edges_neg[1] == 0:
+            bin_edges_neg = bin_edges_neg[1:]
+            bin_counts_neg = bin_counts_neg[1:]
+        return [bin_counts_pos, bin_edges_pos], [bin_counts_neg, bin_edges_neg]
+
+    @property
+    def median_abs_deviation(self):
+        """
+        Get median absolute deviation.
+
+        :return: median absolute deviation
+        """
+        if not self._has_histogram or not self._median_abs_dev_is_enabled:
```

Comment:
will mostly work but will fail on data like: `[-9,-8,4,5,6,7,19,20]` where the median falls between multiple buckets because of averaging between them.

Reply:
agree, added the fix

Reply:
and added the test